lib/middleware/back-links.js

```diff
@@ -12,8 +12,10 @@ module.exports = function backLink(route, controller, steps) {
         var matchingPath = _.find(controller.options.backLinks, function (link) {
             if (link.match(/^\//)) {
                 return path.normalize(link) === referrerPath;
-            } else {
+            } else if (path.relative(baseUrl, referrerPath)) {
                 return path.normalize(link) === path.relative(baseUrl, referrerPath);
+            } else if (referrerPath.replace(/^\//, '') === link.replace(/^\.*\//, '')) {
+                return path.normalize(link);
             }
```

Comment:
"are there any cases where none of the conditions are satisfied?"
Yes, and the function implicitly returns undefined, which is expected